halo2/src/circuit_builder.rs

```diff
@@ -166,6 +170,38 @@ pub(crate) fn analyzed_to_circuit<T: FieldElement>(
             _ => unimplemented!(),
         }
     }
+
+    assert!(
+        analyzed.public_declarations.len() <= num_rows,
+        "More publics than rows!"
+    );
+
+    // Enforce publics by copy-constraining to cells in the instance column.
```

Comment:
Can you add a comment with a small example diagram here of what exactly is happening?

Reply:
Done!

Reply:
where is it?

Reply:
Oh, it got lost somehow? 👀 I just wrote it again, now it's there.